definitions/output/reports/cwv_tech_technologies.js

```diff
@@ -6,19 +6,68 @@ publish('cwv_tech_technologies', {
   tags: ['crux_ready']
 }).query(ctx => `
 /* {"dataform_trigger": "report_cwv_tech_complete", "name": "technologies", "type": "dict"} */
+WITH pages AS (
+  SELECT DISTINCT
+    client,
+    root_page AS origin,
+    tech.technology
+  FROM ${ctx.ref('crawl', 'pages')},
+    UNNEST(technologies) AS tech
+  WHERE
+    date = '${pastMonth}'
+    ${constants.devRankFilter}
+), tech_origins AS (
+  SELECT
+    client,
+    technology,
+    COUNT(origin) AS origins
+  FROM pages
+  GROUP BY
+    client,
+    technology
+), technologies AS (
+  SELECT
+    name AS technology,
+    description,
+    ARRAY_TO_STRING(categories, ', ') AS category,
+    categories AS category_obj,
+    NULL AS similar_technologies
+  FROM ${ctx.ref('wappalyzer', 'apps')}
+), total_pages AS (
+  SELECT
+    client,
+    COUNT(DISTINCT origin) AS origins
+  FROM pages
+  GROUP BY client
+)
+
 SELECT
   client,
-  app AS technology,
+  technology,
   description,
   category,
-  SPLIT(category, ",") AS category_obj,
+  category_obj,
+  similar_technologies,
+  COALESCE(origins, 0) AS origins
+FROM tech_origins
+INNER JOIN technologies
+USING(technology)
+
+UNION ALL
+
+SELECT
+  client,
+  'ALL' AS technology,
+  NULL AS description,
+  ARRAY_TO_STRING(categories, ', ') AS category,
+  categories AS category_obj,
   NULL AS similar_technologies,
   origins
-FROM ${ctx.ref('core_web_vitals', 'technologies')}
-LEFT JOIN ${ctx.ref('wappalyzer', 'apps')}
-ON app = name
-WHERE date = '${pastMonth}' AND
-  geo = 'ALL' AND
-  rank = 'ALL'
-ORDER BY origins DESC
+FROM total_pages
+CROSS JOIN (
+  SELECT
+    ARRAY_AGG(DISTINCT category IGNORE NULLS ORDER BY category) AS categories
+  FROM technologies,
+    UNNEST(category_obj) AS category
+) AS cat
```

Comment:
I'm confused why this is a cross join?

Reply:
left side: 'ALL' technologies, 1 row per client with total origins
right side: list of all categories

Should we set categories list as NULL for such case?

Reply:
Yeah I don't get why you need the categories if you're looking at ALL techs? But happy enough that it's actually a limited CROSS JOIN if there's a good reason for that.

Reply:
Let's have NULL value

Reply:
If you use the category filter to show only technologies within a category, we want to keep the ability to compare with ALL.

<img width="469" alt="image" src="https://github.com/user-attachments/assets/4e3c3867-8dee-4a61-b015-90c7f6d02326" />

In the Looker Studio dashboard, the easiest way to do that was to make ALL a member of all categories. Maybe there's a better way to do that in the new dashboard? If so, it'd require frontend changes.

Reply:
@rviscomi Here the CROSS JOIN (before I removed it) added a list of all categories for "ALL" technology.
I couldn't find a use case on frontend, so removed it.

Your example, if I understand, uses `categories` report for the dropdown, there is a list of technologies per category.
![image](https://github.com/user-attachments/assets/675ebd5d-047f-4657-b476-16be2c0d4993)

Adding a placeholder for aggregate category metrics as 'ALL' would need quite a bit of changes on both back and frontend. Current model works only with technology level metrics. And this change will add something that is category-dependent.

If I understood correctly, it needs to be discussed separately.

Reply:
You're right, I expected it to work the same in the new UI but I guess not. We can easily update the query if we decide to add it back to the UI.